### PR TITLE
[web] Improve CPU usage when building wasm_release

### PR DESCRIPTION
**Pull request by mdebbar**
Doing a `felt build` (without `--host`) causes high cpu usage for some people (including me and @htoor3). This is because our build tries to use `goma` with a high parallelization factor that all ends up happening in the local machine because `goma` doesn't support `emscripten` yet.

This PR disables `goma` for wasm/web builds.

### CODE REVIEW

File: lib/web_ui/dev/build.dart [hunk 1]
```diff
@@ -115,22 +115,29 @@ class GnPipelineStep extends ProcessStep {
...
   @override
   Future<ProcessManager> createProcess() {
     print('Running autoninja...');
     return startProcess(
-      'autoninja',
+      // When build the wasm target that includes CanvasKit, we need to use
+      // `ninja` directly to avoid having a high `-j` value that could
+      // potentially slow down the entire machine because emscripten doesn't use
+      // goma.
+      host ? 'autoninja' : 'ninja',
```
Comment (ditman) [inline lib/web_ui/dev/build.dart:140]:
```dart
'autoninja',
if (!host)
  '--offline',
```

may do the trick? That instructs autoninja to *not* look for goma, so it'll default to whatever J it deems good for local builds.

**Review by ditman — APPROVED** [review]
LGTM

**Review by eyebrowsoffire — CHANGES_REQUESTED** [review]
Actually, maybe the right thing is to actually edit `//flutter/tools/gn`. It has a bunch of stuff to try to determine whether to use goma in there, and maybe we should just short circuit that if we're doing a wasm build.

**Review by ditman — APPROVED** [review]
Works on my machine!

**Review by eyebrowsoffire — APPROVED** [review]
LGTM!

**Comment by Hixie** [comment]
test-exempt: configuration change

**Pull request merged**